enterprise/coderd/provisionerdaemons.go

```diff
@@ -343,19 +343,32 @@ func (*EnterpriseTemplateScheduleStore) SetTemplateScheduleOptions(ctx context.C
 		return tpl, nil
 	}
 
-	template, err := db.UpdateTemplateScheduleByID(ctx, database.UpdateTemplateScheduleByIDParams{
-		ID:                 tpl.ID,
-		UpdatedAt:          database.Now(),
-		AllowUserAutostart: opts.UserAutostartEnabled,
-		AllowUserAutostop:  opts.UserAutostopEnabled,
-		DefaultTTL:         int64(opts.DefaultTTL),
-		MaxTTL:             int64(opts.MaxTTL),
-		FailureTTL:         int64(opts.FailureTTL),
-		InactivityTTL:      int64(opts.InactivityTTL),
-		LockedTTL:          int64(opts.LockedTTL),
-	})
+	var template database.Template
+	err := db.InTx(func(db database.Store) error {
+		err := db.UpdateTemplateScheduleByID(ctx, database.UpdateTemplateScheduleByIDParams{
+			ID:                 tpl.ID,
+			UpdatedAt:          database.Now(),
+			AllowUserAutostart: opts.UserAutostartEnabled,
+			AllowUserAutostop:  opts.UserAutostopEnabled,
+			DefaultTTL:         int64(opts.DefaultTTL),
+			MaxTTL:             int64(opts.MaxTTL),
+			FailureTTL:         int64(opts.FailureTTL),
+			InactivityTTL:      int64(opts.InactivityTTL),
+			LockedTTL:          int64(opts.LockedTTL),
+		})
+		if err != nil {
+			return xerrors.Errorf("update template schedule: %w", err)
+		}
+
+		template, err = db.GetTemplateByID(ctx, tpl.ID)
+		if err != nil {
+			return xerrors.Errorf("get updated template schedule: %w", err)
+		}
+
+		return nil
+	}, nil)
 	if err != nil {
-		return database.Template{}, xerrors.Errorf("update template schedule: %w", err)
+		return database.Template{}, err
 	}
 
 	// Update all workspaces using the template to set the user defined schedule
```

Comment:
Should we also perform the below in the same tx?

Reply:
Probably :+1: